weblate/trans/models/translation.py

```diff
@@ -899,6 +900,19 @@ def merge_upload(self, request, fileobj, overwrite, author=None,
         if filecopy[:3] == codecs.BOM_UTF8:
             filecopy = filecopy[3:]
 
+        # check if the provided file is in an external format
+        # We are able to import different external file formats, but
+        # that needs to convert the external format to a translate-toolkit
+        external_format = external_formats.detect_filename(fileobj.name)
+        if external_format is not None:
+            try:
+                name, content = external_format.convert_to_internal(fileobj.name, filecopy)
+                if name is not None and content is not None:
+                    fileobj.name = name
+                    filecopy = content
```

Comment:
It really looks weird to silently ignore all errors here.

Reply:
I've no clue which errors can occur in `external_format.convert_to_internal` (since that may execute 3rd party code like openpyxl), so I'd rather swallow all exceptions than let weblate crash.
I could log the exception at least (but did not implement that here since there is nothing else logged in the same file...)

as seen in the `try_load` beneath, you also catch all errors from loading the file formats and only raise the last catched exception if nothing could be loaded..
At my code I catch all exceptions because if no external format can be loaded, the internal format could still load that file...

Reply:
At least it should use `weblate.utils.errors.report_error` so that the error appears in Weblate logs and in error monitoring (if enabled).